x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { nullableType } from './lib/nullable';
+import { portSchema } from './lib/schemas';
+import { isOk, promiseResult, Result } from './lib/result_type';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+
+// config definition
+enum WebhookMethods {
+  POST = 'post',
+  PUT = 'put',
+}
+enum WebhookSchemes {
+  HTTP = 'http',
+  HTTPS = 'https',
+}
+
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const HeadersSchema = schema.recordOf(schema.string(), schema.string());
+
+export const CompositeUrlSchema = schema.object({
```

Comment:
I'm not too fussed with the composite URL, as looking at examples online I've yet to find a case where the whole URL isn't good enough.
That said, this feature is already done, I don't feel it adds much cognitive load and isn't blocking us from making feature freeze.

I'll get the other points done and revisit this.

Reply:
I'm honestly wondering if there's a use case tho; I don't think there's anything you can describe with pieces that you can't with a url string, right?

It does add code, and it increases the API surface area that we will have to support forever.

I vote to kick it off the island :-)

Reply:
Yeah, I'm leaning the same way.
Just want to get this PR merged as it's a 3 day weekend in the UK. 

Since the CI takes an hour every time I don't want to push yet another change now. 

I'll remove it on a follow up PR.